lineman/app/components/thread_page/activity_card/activity_card.coffee

```diff
@@ -10,24 +10,23 @@ angular.module('loomioApp').directive 'activityCard', ->
     $scope.lastLoadedSequenceId = 0
     $scope.newActivitySequenceId = $scope.discussion.reader().lastReadSequenceId + 1
     visibleSequenceIds = []
-    rollback = 2
 
     $scope.init = ->
       $scope.discussion.markAsRead(0)
 
       target = _.parseInt($location.hash())
-      if target >= $scope.discussion.firstSequenceId and target < $scope.discussion.lastSequenceId 
-        # valid sequence id is specified in url
-        $scope.initialLoaded  = _.max [target - rollback, 0]
-        $scope.initialFocused = target
-      else if $scope.discussion.isUnread()
+      # if target >= $scope.discussion.firstSequenceId and target < $scope.discussion.lastSequenceId 
+      #   # valid sequence id is specified in url
+      #   $scope.initialLoaded  = _.max [target - rollback, 0]
+      #   $scope.initialFocused = target
+      if $scope.discussion.isUnread()
         # discussion is unread
-        $scope.initialLoaded  = _.max [$scope.discussion.reader().lastReadSequenceId - rollback, 0]
-        $scope.initialFocused = $scope.initialLoaded + rollback
+        $scope.initialLoaded  = _.max [$scope.discussion.reader().lastReadSequenceId - 1, 0]
+        $scope.initialFocused = $scope.discussion.reader().lastReadSequenceId + 1
```

Comment:
either that or you pass a selector (rahter than a sequence id) to
initialFocused and place dom elements into the thread when you pass
lastReadSequenceId

On Wed, 27 May 2015 at 20:06 James Kiesel <notifications@github.com> wrote:

> In lineman/app/components/thread_page/activity_card/activity_card.coffee
> <https://github.com/loomio/loomio/pull/2198#discussion_r31111123>:
>
> >          # discussion is unread
> > -        $scope.initialLoaded  = _.max [$scope.discussion.reader().lastReadSequenceId - rollback, 0]
> > -        $scope.initialFocused = $scope.initialLoaded + rollback
> > +        $scope.initialLoaded  = _.max [$scope.discussion.reader().lastReadSequenceId - 1, 0]
> > +        $scope.initialFocused = $scope.discussion.reader().lastReadSequenceId + 1
>
> Mmmight have to write a little algorithm that's like 'get me the smallest
> sequence id in the record store greater than the lastReadSequenceId'?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/2198/files#r31111123>.
>